src/test/java/uk/gov/pay/adminusers/resources/UserResourceGetMultipleTest.java

```diff
@@ -0,0 +1,141 @@
+                .body("[1]._links", hasSize(1))
+                .body("[1]._links[0].href", is("http://localhost:8080/v1/api/users/" + user2.getExternalId()))
+                .body("[1]._links[0].method", is("GET"))
+                .body("[1]._links[0].rel", is("self"))
+                .body("size()", is(2));
+    }
+
+    @Test
+    public void shouldReturnSingleUser_whenGetUsersWithSingleId() throws Exception {
+        String gatewayAccount1 = valueOf(nextInt());
+        String gatewayAccount2 = valueOf(nextInt());
+        Service service = serviceDbFixture(databaseHelper).withGatewayAccountIds(gatewayAccount1, gatewayAccount2).insertService();
+        String serviceExternalId = service.getExternalId();
+        Role role = roleDbFixture(databaseHelper).insertRole();
+        User user = userDbFixture(databaseHelper).withServiceRole(service.getId(), role.getId()).insertUser();
+
+        givenSetup()
+                .when()
+                .contentType(JSON)
+                .accept(JSON)
+                .get(USERS_RESOURCE +"?ids=" + user.getExternalId())
+                .then()
+                .statusCode(200)
+                .body("[0].external_id", is(user.getExternalId()))
+                .body("[0].username", is(user.getUsername()))
+                .body("[0].password", nullValue())
+                .body("[0].email", is(user.getEmail()))
+                .body("[0].service_roles", hasSize(1))
+                .body("[0].service_roles[0].service.external_id", is(serviceExternalId))
+                .body("[0].service_roles[0].service.name", is(service.getName()))
+                .body("[0].telephone_number", is(user.getTelephoneNumber()))
+                .body("[0].otp_key", is(user.getOtpKey()))
+                .body("[0].login_counter", is(0))
+                .body("[0].disabled", is(false))
+                .body("[0].service_roles[0].role.name", is(role.getName()))
+                .body("[0].service_roles[0].role.description", is(role.getDescription()))
+                .body("[0].service_roles[0].role.permissions", hasSize(role.getPermissions().size()))
+                .body("[0]._links", hasSize(1))
+                .body("[0]._links[0].href", is("http://localhost:8080/v1/api/users/" + user.getExternalId()))
+                .body("[0]._links[0].method", is("GET"))
+                .body("[0]._links[0].rel", is("self"))
+                .body("size()", is(1));
+    }
+
+    @Test
+    public void shouldReturn404_whenGetUsers_withNonExistentExternalIds() throws Exception {
+        givenSetup()
+                .when()
+                .accept(JSON)
+                .get(USERS_RESOURCE +"?ids=NON-EXISTENT-USER,ANOTHER_NON_EXISTENT_USER")
+                .then()
+                .statusCode(404);
+    }
+
+    @Test
+    public void shouldReturn404_whenGetUsers_whereSomeNonExistentExternalIds() throws Exception {
```

Comment:
Nice. Comprehensive in combination with the test above.

Reply:
👍